vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -59,26 +65,419 @@ func extractConversionMetrics(cve cves.CVE5, refs []osvschema.Reference) {
+						addToVersionRange(&versionRange, av.Introduced, "", av.Fixed)
+						versionRanges = append(versionRanges, versionRange)
+					} else if av.Introduced != "" && av.LastAffected != "" {
+						versionRange := osvschema.Range{}
+						addToVersionRange(&versionRange, av.Introduced, av.LastAffected, "")
+						versionRanges = append(versionRanges, versionRange)
+						notes = append(notes, "vers range updated la")
+					}
+				}
+				continue
+			}
+
+			if vers.VersionType == "git" {
+				versionRange := osvschema.Range{}
+				versionRange.Events = append(versionRange.Events, osvschema.Event{
+					Introduced: vers.Version,
+				})
+				versionRanges = append(versionRanges, versionRange)
+				continue
+			}
+
+			// Try to extract versions from text like "before 1.4.7".
+			possibleVersions, note := cves.ExtractVersionsFromText(nil, vers.Version)
+			if note != nil {
+				notes = append(notes, note...)
+			}
+			if possibleVersions != nil {
+				notes = append(notes, "Versions retrieved from text but not used CURRENTLY")
+				continue
+			}
+
+			// As a fallback, assume a single version means it's the fixed version.
+			if vQuality <= vulns.Spaces {
+				versionRange := osvschema.Range{}
+				addToVersionRange(&versionRange, "0", "", vers.Version)
+				versionRanges = append(versionRanges, versionRange)
+				notes = append(notes, fmt.Sprintf("%s - Single version found %v - Assuming introduced = 0 and Fixed = %v", vQuality, vers.Version, vers.Version))
+			}
+		}
+	}
+	// Determine the most frequent version type to return as the range type.
+	maxCount := 0
+	var mostFrequentVersionType string
+	for versionType, count := range versionTypesCount {
+		if count > maxCount {
+			maxCount = count
+			mostFrequentVersionType = versionType
+		}
+	}
+
+	return versionRanges, mostFrequentVersionType, notes
+}
+
+// sortBadSemver provides a custom sorting function for version strings that may not
+// strictly adhere to the SemVer specification. It compares versions numerically,
+// part by part (major, minor, patch).
+func sortBadSemver(a, b string) int {
+	partsA := strings.Split(a, ".")
+	partsB := strings.Split(b, ".")
+	majorA, _ := strconv.Atoi(partsA[0])
+	majorB, _ := strconv.Atoi(partsB[0])
+
+	if c := cmp.Compare(majorA, majorB); c != 0 {
+		return c
+	}
+
+	minorA, _ := strconv.Atoi(partsA[1])
+	minorB, _ := strconv.Atoi(partsB[1])
+	if c := cmp.Compare(minorA, minorB); c != 0 {
+		return c
+	}
+	patchA, _ := strconv.Atoi(partsA[2])
+	patchB, _ := strconv.Atoi(partsB[2])
+	return cmp.Compare(patchA, patchB)
+}
+
+// findInverseAffectedRanges calculates the affected version ranges by analyzing a list
+// of 'unaffected' versions. This is common in Linux kernel CVEs where a product is
+// considered affected by default, and only unaffected versions are listed.
+// It sorts the introduced and fixed versions to create chronological ranges.
+func findInverseAffectedRanges(cveAff cves.Affected) (ranges []osvschema.Range, notes []string) {
```

Comment:
This function seems pretty specific to Linux CNA published CVEs, can we add a check for the CNA if we are only currently Linux CNA, and don't convert non linux CNA records here.